libs/ui/src/ui_strings/play_audio_clips.tsx

```diff
@@ -1,23 +1,124 @@
-/* istanbul ignore file - pending implementation. */
-
 import React from 'react';
 
 import { LanguageCode } from '@votingworks/types';
+import { assertDefined } from '@votingworks/basics';
+
+import { useAudioContext } from './audio_context';
+import { AudioPlayer, newAudioPlayer } from './audio_player';
 
 export interface ClipParams {
   audioId: string;
   languageCode: LanguageCode;
 }
 
-export interface PlayAudioClipsProps {
+type PlayAudioClipProps = ClipParams & {
+  onDone: () => void;
+};
+
+function PlayAudioClip(props: PlayAudioClipProps) {
+  const { audioId, languageCode, onDone } = props;
+  const [audioPlayer, setAudioPlayer] = React.useState<AudioPlayer>();
+  const { api, gainDb, playbackRate, webAudioContext } = assertDefined(
+    useAudioContext()
+  );
+
+  const audioOutputRef = React.useRef<AudioPlayer>();
+  const gainDbRef = React.useRef<number>(gainDb);
+  const onDoneRef = React.useRef(onDone);
+  const playbackRateRef = React.useRef<number>(playbackRate);
```

Comment:
For my understanding, is the usage of refs throughout this component to allow for their use in effects without having the effects update in response to changes to them? A quick comment noting that could be helpful if so

Reply:
Good call -- will add a comment. Might also be able to re-organise the audio player interface a bit to make some of these unnecessary.